### PR TITLE
Add support for sub-title/actor/director/writer

### PR DESCRIPTION
**Pull request by ghost**
v3.6.0

Add support for episode-name/cast/director/writer

### CODE REVIEW

File: src/PVRIptvData.cpp [hunk 1]
```diff
@@ -802,9 +810,9 @@ PVR_ERROR PVRIptvData::GetEPGForChannel(ADDON_HANDLE handle, const PVR_CHANNEL &
       tag.strPlotOutline      = myTag->strPlotOutline.c_str();
       tag.strPlot             = myTag->strPlot.c_str();
       tag.strOriginalTitle    = NULL;  /* not supported */
-      tag.strCast             = NULL;  /* not supported */
-      tag.strDirector         = NULL;  /* not supported */
-      tag.strWriter           = NULL;  /* not supported */
+      tag.strCast             = myTag->strActor.c_str();
```
Comment (phunkyfish) [inline src/PVRIptvData.cpp]: Do you have a sample of the XML used for this. Is this a single actor or a list of actors?

File: src/PVRIptvData.cpp [hunk 2]
```diff
@@ -826,7 +834,7 @@ PVR_ERROR PVRIptvData::GetEPGForChannel(ADDON_HANDLE handle, const PVR_CHANNEL &
       tag.iSeriesNumber       = 0;     /* not supported */
       tag.iEpisodeNumber      = 0;     /* not supported */
       tag.iEpisodePartNumber  = 0;     /* not supported */
-      tag.strEpisodeName      = NULL;  /* not supported */
+      tag.strEpisodeName      = myTag->strSubTitle.c_str();
```
Comment (phunkyfish) [inline src/PVRIptvData.cpp]: Would be better to name the strSubTitle to strEpisodeName

File: src/PVRIptvData.h [hunk 3]
```diff
@@ -38,10 +38,14 @@ struct PVRIptvEpgEntry
   time_t      startTime;
   time_t      endTime;
   std::string strTitle;
+  std::string strSubTitle;
```
Comment (phunkyfish) [inline src/PVRIptvData.h]: Rename to strEpisodeTitle

**Comment by phunkyfish** [comment]
Note that you need to add an entry to the changelog and increase the minor version number. In addition you also need to create a PR for the Matrix branch and get that merged first. 

CI for Leia is currently broken so you once this won't be able to go in before is fixed.

File: src/PVRIptvData.cpp (hunk 1 above)
Comment (ghost) [reply src/PVRIptvData.cpp]: This is one string with list of the actors separated with delimiter it's look nicer.
it's like category element, can be more that one, but used / readed only first, then better concatenate element value with delimiter

```xml
<programme channel="lietuvos-rytas-tv" start="20190815110500 +0300" stop="20190815121000 +0300">
<title>Šeimininkė</title>
<sub-title>1-as sezonas. 4-a serija</sub-title>
<desc>Alina – šeimininkė. Ji valdo visą šachtininkų miestelį...</desc>
<category>Movie / Drama</category>
<icon src="https://cdn.tvstart.com/img/programme/thumb/380708.jpg" />
<credits>
<director>Bata Nedich / Other Director</director>
<writer>Writer</writer>
<actor>Ekaterina Varchenko / Elena Stefanskaja / Sergej Radchenko</actor>
</credits>
</programme>
```

screenshot007
screenshot006

File: src/PVRIptvData.cpp (hunk 2 above)
Comment (ghost) [reply src/PVRIptvData.cpp]: ok

File: src/PVRIptvData.h (hunk 3 above)
Comment (ghost) [reply src/PVRIptvData.h]: ok

**Comment by ghost** [comment]
I updated PR for Leia branch

File: src/PVRIptvData.cpp (hunk 1 above)
Comment (phunkyfish) [reply src/PVRIptvData.cpp]: Ok, maybe also rename strActor as strCast. Similar to strEpisode name.

File: README.md [hunk 4]
```diff
@@ -10,7 +10,7 @@ IPTV Live TV and Radio PVR client addon for [Kodi](http://kodi.tv)
 ### Linux
 
 1. `git clone --branch Leia https://github.com/xbmc/xbmc.git`
-2. `git clone https://github.com/kodi-pvr/pvr.iptvsimple.git`
+2. `git clone --branch Leia https://github.com/kodi-pvr/pvr.iptvsimple.git`
```
Comment (phunkyfish) [inline README.md]: Unrelated change. Should be in separate commit.

File: pvr.iptvsimple/addon.xml.in [hunk 5]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <addon
   id="pvr.iptvsimple"
-  version="3.5.8"
+  version="3.5.9"
```
Comment (phunkyfish) [inline pvr.iptvsimple/addon.xml.in]: It’s major.minor.patch so should be 3.6.0

File: pvr.iptvsimple/changelog.txt [hunk 6]
```diff
@@ -1,3 +1,6 @@
+v3.5.9
```
Comment (phunkyfish) [inline pvr.iptvsimple/changelog.txt]: Same here

File: README.md (hunk 4 above)
Comment (ghost) [reply README.md]: ok

File: pvr.iptvsimple/addon.xml.in (hunk 5 above)
Comment (ghost) [reply pvr.iptvsimple/addon.xml.in]: updated

File: pvr.iptvsimple/changelog.txt (hunk 6 above)
Comment (ghost) [reply pvr.iptvsimple/changelog.txt]: updated

File: src/PVRIptvData.cpp (hunk 1 above)
Comment (ghost) [reply src/PVRIptvData.cpp]: I changed as strCast

**Comment by phunkyfish** [comment]
You can still add the Readme change in this PR, just needs to be a separate commit.

**Comment by ghost** [comment]
Added fix readme in the separate commit

**Comment by phunkyfish** [comment]
Cool, now you need to create another PR for Matrix which will be similar to this. Note that version number will be different.

**Comment by phunkyfish** [comment]
Thanks for following up on the notes. Hopefully I will merge the PR for Matrix later today but this will have to wait until Jenkins is fixed for Leia. Not sure when that will be but will circle back once it's ready.

File: pvr.iptvsimple/changelog.txt [hunk 7]
```diff
@@ -1,3 +1,6 @@
+v3.6.0
+- Add support for sub-title/actor/director/writer in XML
```
Comment (phunkyfish) [inline pvr.iptvsimple/changelog.txt]: Should read: `Add support for episode-name/cast/director/writer"

Comment (phunkyfish) [reply pvr.iptvsimple/changelog.txt]: Same with commit name

**Comment by ghost** [comment]
For me it's was problem with xbmc p8-platform not existing commit, I changed it and so solve problem

**Comment by phunkyfish** [comment]
Same here, but for Jenkins it needs to solved properly which could take a bit of time.

**Comment by phunkyfish** [comment]
Once you make the change above it should build as correctly on jenkins as that appears fixed now.

File: pvr.iptvsimple/changelog.txt (hunk 7 above)
Comment (ghost) [reply pvr.iptvsimple/changelog.txt]: sorry, my mistake, updateted

**Review by phunkyfish — APPROVED** [review]
Looks good

**Comment by phunkyfish** [comment]
@2small do you have a sample m3u that you used for this?

**Comment by ghost** [comment]
> @2small do you have a sample m3u that you used for this?

```
#EXTM3U
#EXTINF:-1,LRT Televizija
http://localhost/stream/channelid/753198401
```
I don't use `tvg-name` or `tvg-id`, because is not necessary

I see one problem, my xmltv and `pvr.iptvsimple` client didn't work with list of `display-name`, only first:
```xml
<channel id="lrt-televizija">
    <display-name>LRT Televizija</display-name>
    <display-name>LRT Televizija HD</display-name>
</channel>
```

Then, in the m3u, need describe `tvg-name` or `tvg-id`
```
#EXTM3U
#EXTINF:-1,LRT Televizija
http://localhost/stream/channelid/753198401
#EXTINF:-1 tvg-name="LRT Televizija",LRT Televizija HD
http://localhost/stream/channelid/1564369636
```
But I have mapper and map both same channels to one with priority `HD`, but `HD` ending simple stripped

**Comment by phunkyfish** [comment]
Sorry, I mean an example of the xml for actor/director/writer

**Comment by phunkyfish** [comment]
For the issue you've pointed out above, please create an issue. A PR for a different purpose is not the right place to discuss it. Thanks ;)

**Comment by ghost** [comment]
> For the issue you've pointed out above, please create an issue. A PR for a different purpose is not the right place to discuss it. Thanks ;)

Yes I know, but need brainstorming for this change, maybe later :)
I send email with sample xmltv link

**Comment by phunkyfish** [comment]
:)

I got the sample, thank you. It appears there are two formats for actor:

1. One where they are all on one line which your solution works for
2. Another where there are multiple actor elements. In this case your solution will only pick the first single actor.

Example:

```
		<credits>
			<director>Justin Lin</director>
			<actor>Vin Diesel</actor>
			<actor>Paul Walker</actor>
			<actor>Dwayne Johnson</actor>
			<actor>Jordana Brewster</actor>
			<actor>Michelle Rodriguez</actor>
			<actor>Tyrese Gibson</actor>
			<actor>Sung Kang</actor>
			<actor>Gal Gadot</actor>
			<actor>Ludacris</actor>
			<actor>Luke Evans</actor>
			<actor>Elsa Pataky</actor>
			<actor>Gina Carano</actor>
			<actor>Clara Paget</actor>
			<actor>Kim Kold</actor>
			<actor>Joe Taslim</actor>
			<actor />
		</credits>
```

**Comment by ghost** [comment]
yes, but if we have list element of `actor`
Kodi display this actors list separete with new line
and the show info dialog window we have about 10 lines content
then better join actors to one element, then we can see more info

https://2small.eu/wp-content/uploads/2019/08/programos-informacija-1024x576.png

otherwise if have 2-3 `director` and 10-15 `actor` you can't see full list of this, because action is only full `description` about show

it's same with `category` tag use only first not list `category` tags

**Comment by phunkyfish** [comment]
What about combining multiple into one?

**Comment by ghost** [comment]
Maybe can give me sample code how to do (read/foreach/combine)
because with `cpp` don't programming :( java/go/python :)
And I can make change then.

**Comment by phunkyfish** [comment]
Might be easier if I crate a temp PR you can test and modify. As I don’t have M3U’s that have multiple actors you may need to fix it a bit.

**Comment by ghost** [comment]
OK, you can simple make with one tag, others I can do
